src/main/java/uk/gov/hmcts/reform/preapi/media/AzureMediaService.java

```diff
@@ -61,22 +75,135 @@ public List<AssetDTO> getAssets() {
+                                    ))
+                                    .withDescription(
+                                        "Streaming Endpoint for " + streamingEndpointName
+                                    )
+                            );
+        } catch (ManagementException e) {
+            if (e.getResponse().getStatusCode() == 409) {
+                return amsClient.getStreamingEndpoints()
+                                .get(resourceGroup, accountName, streamingEndpointName);
+            }
+            throw e;
+        }
     }
 
+    private void assertStreamingLocatorExists(UUID liveEventId) {
+
+        try {
+            Logger.getAnonymousLogger().info("Creating Streaming locator");
+            var sanitisedLiveEventId = getSanitisedLiveEventId(liveEventId);
+            var streamingLocatorProperties = new StreamingLocatorInner()
+                .withAssetName(sanitisedLiveEventId)
+                .withStreamingPolicyName("Predefined_ClearStreamingOnly")
+                .withStreamingLocatorId(liveEventId);
+
+            amsClient.getStreamingLocators().create(resourceGroup,
+                                                    accountName,
+                                                    sanitisedLiveEventId,
+                                                    streamingLocatorProperties);
+        } catch (ManagementException e) {
+            if (e.getResponse().getStatusCode() == 409) {
+                Logger.getAnonymousLogger().info("Streaming locator already exists");
+                return;
+            }
+            throw e;
+        }
+    }
+
+    private void assertLiveEventExists(@NotNull UUID liveEventId) {
+        var sanitisedLiveEventId = getSanitisedLiveEventId(liveEventId);
+        try {
+            var liveEvent = amsClient.getLiveEvents().get(resourceGroup, accountName, sanitisedLiveEventId);
+            if (!liveEvent.resourceState().equals(LiveEventResourceState.RUNNING)) {
+                throw new AMSLiveEventNotRunningException(sanitisedLiveEventId);
+            }
+        } catch (ManagementException e) {
+            if (e.getResponse().getStatusCode() == 404) {
+                throw new AMSLiveEventNotFoundException(sanitisedLiveEventId);
+            }
+            throw e;
+        }
+    }
+
+    private String parseLiveOutputUrlFromStreamingLocatorPaths(String streamingEndpointHostname,
+                                                               @NotNull ListPathsResponseInner paths) {
+        Logger.getAnonymousLogger().info("parsing live output url from streaming locator paths");
+        Logger.getAnonymousLogger().info(streamingEndpointHostname);
+        paths.streamingPaths().forEach(p -> Logger.getAnonymousLogger().info(p.paths().toString()));
+        return paths.streamingPaths().stream()
+            .flatMap(path -> path.paths().stream())
+            .map(p -> "https://" + streamingEndpointHostname + p)
+            .findFirst()
```

Comment:
Could change this so you only map one string
```suggestion
            .findFirst()
            .map(p -> "https://" + streamingEndpointHostname + p)
```